File: solutions/python/perfect-numbers/1/perfect_numbers.py
def get_factors(number):
    factors = []
    for num in range(1, number // 2 + 1):
        if number % num == 0:
            factors.append(num)
    return factors

def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")
    
    factors = get_factors(number)
    sum = 0
    for factor in factors:
        sum += factor

    if number < sum:
        return "abundant"
    elif number > sum:
        return "deficient"
    return "perfect"

File: solutions/csharp/perfect-numbers/1/PerfectNumbers.cs
public enum Classification
{
  Perfect,
  Abundant,
  Deficient
}

public static class PerfectNumbers
{
  public static Classification Classify(int number)
  {
    if (number < 1)
    {
      throw new ArgumentOutOfRangeException("number", "Classification is only possible for natural numbers.");
    }

    var factors = GetFactors(number);
    var sumOfFactors = factors.Sum();
    if (number == sumOfFactors)
    {
      return Classification.Perfect;
    }
    else if (number < sumOfFactors)
    {
      return Classification.Abundant;
    }
    return Classification.Deficient;
  }

  private static IEnumerable<int> GetFactors(int number)
  {
    var factors = new List<int>();
    for (int i = 1; i <= number / 2; i++)
    {
      if (number % i == 0)
      {
        yield return i;
      }
    }
  }
}
